src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,272 @@
+  networks: Networks;
+}) {
+  const { currency } = useCurrency();
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+
+  const handleDialogOpen = useCallback(() => {
+    dialogRef.current?.showModal();
+  }, []);
+
+  const handleDialogDismiss = useCallback(() => {
+    dialogRef.current?.close();
+  }, []);
+
+  const { transfer, isIncoming } = useMemo(() => {
+    const incomingTransfers = action.content?.transfers?.incoming?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedIncomingTransfer: ActionTransfer | undefined =
+      incomingTransfers?.length
+        ? {
+            ...incomingTransfers[0],
+            quantity: incomingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    const outgoingTransfers = action.content?.transfers?.outgoing?.filter(
+      (item) => getFungibleAsset(item.asset)?.id === asset.id
+    );
+    const aggregatedOutgoingTransfer: ActionTransfer | undefined =
+      outgoingTransfers?.length
+        ? {
+            ...outgoingTransfers[0],
+            quantity: outgoingTransfers
+              .reduce((acc, item) => acc.plus(item.quantity), new BigNumber(0))
+              .toFixed(),
+          }
+        : undefined;
+    return {
+      transfer: aggregatedIncomingTransfer || aggregatedOutgoingTransfer,
+      isIncoming: Boolean(aggregatedIncomingTransfer),
+    };
+  }, [action, asset.id]);
+
+  const fungible = getFungibleAsset(transfer?.asset);
+
+  if (!fungible || !transfer) {
+    return null;
+  }
+
+  const actionType =
+    action.type.value === 'trade'
+      ? isIncoming
+        ? 'Buy'
+        : 'Sell'
+      : action.type.display_value;
+
+  const formattedPrice = transfer.price
+    ? formatPriceValue(transfer.price, 'en', currency)
+    : noValueDash;
+
+  const normalizedQuantity = baseToCommon(
+    transfer.quantity,
+    getDecimals({
+      asset: fungible,
+      chain: createChain(action.transaction.chain),
+    })
+  );
+
+  const normalizedAmountAction =
+    transfer.price === null || transfer.price === undefined
+      ? null
+      : normalizedQuantity.multipliedBy(transfer.price);
+
+  const actionTitle = `${actionType} at ${formattedPrice}`;
+  const actionDatetime = dateFormatter.format(new Date(action.datetime));
+  const actionBalance = `${isIncoming ? '+' : minus}${formatTokenValue(
+    normalizedQuantity,
+    asset.symbol,
+    { notation: normalizedQuantity.gte(100000) ? 'compact' : undefined }
+  )}`;
+  const actionValue = normalizedAmountAction
+    ? formatCurrencyValue(normalizedAmountAction, 'en', currency)
+    : null;
+
+  return (
+    <>
+      <KeyboardShortcut
+        combination="backspace"
+        onKeyDown={handleDialogDismiss}
+      />
+      <UnstyledButton onClick={handleDialogOpen} className={styles.historyItem}>
+        <div className={styles.historyItemBackdrop} />
```

Comment:
Why not reuse `PageFullBleed` + `SurfaceList`? Then you wouldn't need this helper

Reply:
Good idea, let me try!

Reply:
I fixed this partially. 
Using SurfaceList is hard the same way it was in the main History.
Because of the Dialog associated with every item of the list, it is hard to use onClick outside of the component, so by default items will not be wrapped with the decorations which is the main point of the SurfaceList at this point.
+ It looks like we will need to update this list soon too and design will be updated significantly